src/validator.cpp

```diff
@@ -828,27 +799,18 @@ void Validator::ValidatorImpl::validateMath(const std::string &input, const Comp
     }
 
     validateMathMLElements(nodeCopy, component);
-    // Get the bvar names in this math element.
-    // TODO: may want to do this with XPath instead...
-    gatherMathBvarVariableNames(nodeCopy, bvarNames);
-    // Check that no variable names match new bvar names.
-    for (const std::string &variableName : variableNames) {
-        if (std::find(bvarNames.begin(), bvarNames.end(), variableName) != bvarNames.end()) {
-            ErrorPtr err = std::make_shared<Error>();
-            err->setDescription("Math in component '" + component->name() + "' contains '" + variableName + "' as a bvar ci element but it is already a variable name.");
-            err->setComponent(component);
-            err->setKind(Error::Kind::MATHML);
-            mValidator->addError(err);
-        }
-    }
+
     // Iterate through ci/cn elements and remove cellml units attributes.
     XmlNodePtr mathNode = node;
-    validateAndCleanMathCiCnNodes(node, component, variableNames, bvarNames);
-    // Get the MathML string (with cellml:units attributes already removed) and remove the CellML namespace.
-    // While the removeSubstring() approach for removing the cellml namespace before validating with the MathML DTD
-    // is not ideal, libxml does not appear to have a better way to remove a namespace declaration from the tree.
+    validateAndCleanMathCiCnNodes(node, component, variableNames);
+
+    // Remove the cellml namespace definition.
+    if (mathNode->hasNamespaceDefinition(CELLML_2_0_NS)) {
```

Comment:
Because this is about to be passed to the MathML DTD and if we have a stand alone math document it is likely to have a `cellml` namespace definition here.